lib/AST/Decl.cpp

```diff
@@ -6021,7 +6021,18 @@ bool ConstructorDecl::isObjCZeroParameterWithLongSelector() const {
   if (params->size() != 1)
     return false;
 
-  return params->get(0)->getInterfaceType()->isVoid();
+  if (params->get(0)->hasInterfaceType()) {
+    return params->get(0)->getInterfaceType()->isVoid();
+  } else if (auto tyR = params->get(0)->getTypeLoc().getTypeRepr()) {
+    // Fallback to textual check. This can happen when getting ObjC selector
```

Comment:
I would prefer it if instead you did

```
if (auto *resolver = getASTContext().getLazyResolver())
  resolver->resolveDeclSignature(this);
```

Reply:
Is there really a resolver ? If there is a resolver it seems like a mistake to me because we don't need any kind of typechecking for syntactic requests so we shouldn't create a resolver.

Reply:
Confirmed there's no resolver at this point.